src/lib/utils/__tests__/mergeSearchParameters-test.ts

```diff
@@ -0,0 +1,370 @@
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'colors'],
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'categories', 'colors'],
+      })
+    );
+  });
+
+  it('merges `disjunctiveFacets` parameters', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['categories'],
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand', 'colors'],
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand', 'categories', 'colors'],
+      })
+    );
+  });
+
+  it('merges `facetsRefinements` parameters, overrides conflicts', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand'],
+        facetsRefinements: {
+          brand: ['Samsung'],
+        },
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['categories'],
+        facetsRefinements: {
+          categories: ['TVs'],
+        },
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'colors'],
+        facetsRefinements: {
+          brand: ['Apple'],
+          colors: ['Red'],
+        },
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'categories', 'colors'],
+        facetsRefinements: {
+          brand: ['Apple'],
+          categories: ['TVs'],
+          colors: ['Red'],
+        },
+      })
+    );
+  });
+
+  it('merges `facetsExcludes` parameters, overrides conflicts', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand'],
+        facetsExcludes: {
+          brand: ['Samsung'],
+        },
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['categories'],
+        facetsExcludes: {
+          categories: ['TVs'],
+        },
+      }),
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'colors'],
+        facetsExcludes: {
+          brand: ['Apple'],
+          colors: ['Red'],
+        },
+      })
+    );
+
+    expect(actual).toEqual(
+      algoliasearchHelper.SearchParameters.make({
+        facets: ['brand', 'categories', 'colors'],
+        facetsExcludes: {
+          brand: ['Apple'],
+          categories: ['TVs'],
+          colors: ['Red'],
+        },
+      })
+    );
+  });
+
+  it('merges `disjunctiveFacetsRefinements` parameters, overrides conflicts', () => {
+    const actual = merge(
+      algoliasearchHelper.SearchParameters.make({
+        disjunctiveFacets: ['brand'],
+        disjunctiveFacetsRefinements: {
+          brand: ['Samsung'],
```

Comment:
Since we now consider default `disjunctivefacetsRefinements` to be empty arrays (e.g. algolia/algoliasearch-helper-js#725), we should add empty arrays in our test cases (valid for all array refinements).

Reply:
The change you've pointed at does not impact the function. The default value is only returned by the getter, it's not stored like this on the `SearchParameters`. We don't use getter in this situation we use the "raw" attributes of `SearchParameters` which does not contain default values for each attributes.

Reply:
Ah, right!